src/main/java/nl/ou/debm/common/feature3/FunctionProducer.java

```diff
@@ -36,15 +38,44 @@ public FunctionProducer(CGenerator generator){
 
     @Override
     public String getNewExpression(int currentDepth, DataType type, boolean terminating) {
+        if(currentDepth < 100 && generator.functions.size() < FUNCTIONS_MIN)
+            return getFunctionCall(currentDepth + 1, type);
+        if(currentDepth < 50 && terminating && !isSatisfied()) {
+            var x = getFunctionCall(currentDepth + 1, type);
+            if(x.startsWith("function_")){
+                System.out.println("x");
+            }
+            return x;
+        }
+
         if(terminating || Math.random() < 0.6){
             return type.strDefaultValue(generator.structsByName);
         }else{
-            return getFunctionCall(currentDepth + 1, type, null);
+            var x = getFunctionCall(currentDepth + 1, type);
+            if(x.startsWith("function_")){
+                System.out.println("x");
```

Comment:
This looks like a debug hook. Should it be removed? Or turned into an assert/throw RuntimeException?